akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -36,12 +38,43 @@ trait Projection[Envelope] {
 
   def projectionId: ProjectionId
 
-  def withSettings(settings: ProjectionSettings): Projection[Envelope]
+  def withRestartBackoff(
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double): Projection[Envelope]
+
+  def withRestartBackoff(
+      minBackoff: FiniteDuration,
+      maxBackoff: FiniteDuration,
+      randomFactor: Double,
+      maxRestarts: Int): Projection[Envelope]
+
+  /**
+   * Java API
+   */
+  def withRestartBackoff(
+      minBackoff: java.time.Duration,
+      maxBackoff: java.time.Duration,
+      randomFactor: Double): Projection[Envelope]
+
+  /**
+   * Java API
+   */
+  def withRestartBackoff(
+      minBackoff: java.time.Duration,
+      maxBackoff: java.time.Duration,
+      randomFactor: Double,
+      maxRestarts: Int): Projection[Envelope]
 
   def statusObserver: StatusObserver[Envelope]
 
   def withStatusObserver(observer: StatusObserver[Envelope]): Projection[Envelope]
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def withSettings(settings: ProjectionSettings): Projection[Envelope]
```

Comment:
This will be visible from Java, no? We could have a `ProjectionInternal` trait that are implemented by the Impl, so we hide it in the `Projection` trait. 

Or, we could simply remove it. The only place we are using it is in the testkit. Instead, we can use `withRestartBackoff` directly. 

(We can explore it on another PR, so we don't block the release.)

Reply:
Unrelated question. 

We are using a mix of `private[akka]`, `private[projection]` and `private[internal]`. Should we align it? And if so, which one to pick? I would say, `private[projection]`. WDYT?

Reply:
> This will be visible from Java, no? We could have a ProjectionInternal trait that are implemented by the Impl, so we hide it in the Projection trait.

It will be visible for Java, but we already have several more such internal methods there. I don't think we can completely hide them. We could have only one method and access rest from there, but probably not a big api problem.

> We are using a mix of private[akka], private[projection] and private[internal]

Doesn't matter much. I often run into compiler warnings when using to restricted scope, when something is exposed by something else with different scope. More trouble than it's worth. That's why I often go with `akka`, but `projection` would probably work.